google/utils.go

```diff
@@ -387,3 +363,274 @@ func retry(retryFunc func() error) error {
+		if (network == "" && subnetwork == "") || (network != "" && subnetwork != "") {
+			return nil, fmt.Errorf("exactly one of network or subnetwork must be provided")
+		}
+
+		networkLink, err := getNetworkLink(config, project, network)
+		if err != nil {
+			return nil, fmt.Errorf("cannot determine selflink for subnetwork '%s': %s", subnetwork, err)
+		}
+
+		subnetworkProject := data["subnetwork_project"].(string)
+		subnetLink, err := getSubnetworkLink(config, project, region, subnetworkProject, subnetwork)
+		if err != nil {
+			return nil, fmt.Errorf("cannot determine selflink for subnetwork '%s': %s", subnetwork, err)
+		}
+
+		ifaces[i] = &compute.NetworkInterface{
+			NetworkIP:     data["network_ip"].(string),
+			Network:       networkLink,
+			Subnetwork:    subnetLink,
+			AccessConfigs: expandAccessConfigs(data["access_config"].([]interface{})),
+			AliasIpRanges: expandAliasIpRanges(data["alias_ip_range"].([]interface{})),
+		}
+
+		// network_ip is deprecated. We want address to win if both are set.
+		if data["address"].(string) != "" {
+			ifaces[i].NetworkIP = data["address"].(string)
+		}
+
+	}
+	return ifaces, nil
+}
+
+func flattenServiceAccounts(serviceAccounts []*compute.ServiceAccount) []map[string]interface{} {
+	result := make([]map[string]interface{}, len(serviceAccounts))
+	for i, serviceAccount := range serviceAccounts {
+		result[i] = map[string]interface{}{
+			"email":  serviceAccount.Email,
+			"scopes": schema.NewSet(stringScopeHashcode, convertStringArrToInterface(serviceAccount.Scopes)),
+		}
+	}
+	return result
+}
+
+func expandServiceAccounts(configs []interface{}) []*compute.ServiceAccount {
+	accounts := make([]*compute.ServiceAccount, len(configs))
+	for i, raw := range configs {
+		data := raw.(map[string]interface{})
+
+		accounts[i] = &compute.ServiceAccount{
+			Email:  data["email"].(string),
+			Scopes: canonicalizeServiceScopes(convertStringSet(data["scopes"].(*schema.Set))),
+		}
+
+		if accounts[i].Email == "" {
+			accounts[i].Email = "default"
+		}
+	}
+	return accounts
+}
+
+func expandGuestAccelerators(zone string, configs []interface{}) []*compute.AcceleratorConfig {
+	guestAccelerators := make([]*compute.AcceleratorConfig, len(configs))
+	for i, raw := range configs {
+		data := raw.(map[string]interface{})
+		guestAccelerators[i] = &compute.AcceleratorConfig{
+			AcceleratorCount: int64(data["count"].(int)),
+			AcceleratorType:  acceleratorRef(zone, data["type"].(string)),
+		}
+	}
+
+	return guestAccelerators
+}
+
+func flattenGuestAccelerators(accelerators []*compute.AcceleratorConfig) []map[string]interface{} {
+	acceleratorsSchema := make([]map[string]interface{}, len(accelerators))
+	for i, accelerator := range accelerators {
+		acceleratorsSchema[i] = map[string]interface{}{
+			"count": accelerator.AcceleratorCount,
+			"type":  accelerator.AcceleratorType,
+		}
+	}
+	return acceleratorsSchema
+}
+
+// Instances want a partial URL, but instance templates want the bare
```

Comment:
You can use the pattern we are using here: https://github.com/terraform-providers/terraform-provider-google/blob/master/google/field_helpers.go#L20

This way both instance and instance template can accept either the name or the URL. And then when calling the api, you can pass whichever the API requires.